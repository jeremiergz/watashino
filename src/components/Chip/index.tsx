import Text, { TextProps } from 'components/primitives/Text';
import { useTheming } from 'components/providers/ThemeProvider';
import React from 'react';
import { lightenColor } from 'utils/CSS';

const Chip: React.FC<ChipProps> = ({ children, color, ...rest }) => {
  const { theme } = useTheming();
  const fontColor = color || theme.colors.gray;
  const backgroundColor = lightenColor(fontColor, 0.25);
  return (
    <Text
      as="span"
      backgroundColor={backgroundColor}
      borderRadius={12}
      color={fontColor}
      fontSize={12}
      fontWeight="semiBold"
      lineHeight="12px"
      maxHeight={12}
      paddingBottom={1}
      paddingTop="3px"
      paddingX={2}
      {...rest}
    >
      {children}
    </Text>
  );
};

Chip.displayName = 'Chip';

export type ChipProps = Omit<TextProps, 'color'> & {
  color?: string;
};
export default Chip;
