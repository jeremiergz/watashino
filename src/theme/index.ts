const breakpoints = ['320px', '375px', '425px', '768px', '1024px', '1200px', '1440px', '2560px'] as const;
Object.defineProperties(breakpoints, {
  mobileS: { value: breakpoints[0] },
  mobileM: { value: breakpoints[1] },
  mobileL: { value: breakpoints[2] },
  tablet: { value: breakpoints[3] },
  laptopS: { value: breakpoints[4] },
  laptopM: { value: breakpoints[5] },
  laptopL: { value: breakpoints[6] },
  desktop: { value: breakpoints[7] },
});

const colors = {
  black: '#212121',
  dark: '#424242',
  error: '#F44336',
  gray: '#757575',
  info: '#2196F3',
  lightgray: '#E0E0E0',
  primary: '#31859A',
  secondary: '#17365C',
  success: '#4CAF50',
  white: '#FAFAFA',
} as const;

const fonts = {
  main: "'Open Sans', sans-serif",
} as const;

const fontWeights = {
  bold: 800,
  regular: 400,
  'semi-bold': 600,
} as const;

const DarkTheme = {
  background: colors.black,
  breakpoints,
  colors: {
    ...colors,
    text: colors.white,
  },
  fonts,
  fontWeights,
  svg: colors.white,
  type: 'dark',
};

const LightTheme = {
  background: colors.white,
  breakpoints,
  colors: {
    ...colors,
    text: '#262626',
  },
  fonts,
  fontWeights,
  svg: '#262626',
  type: 'light',
};

type Theme = typeof DarkTheme | typeof LightTheme;
type ThemeProps = {
  background?: string;
  colors?: keyof Theme['colors'] | 'inherit';
  fontWeight?: keyof Theme['fontWeights'];
  svg?: string;
  type?: 'light' | 'dark';
};

export { colors, DarkTheme, LightTheme, Theme, ThemeProps };
