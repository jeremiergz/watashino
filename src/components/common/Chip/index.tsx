import React from 'react';
import useTheme from '../../../hooks/useTheme';
import { Theme } from '../../../theme';
import { lightenColor } from '../../../utils/CSS';
import Text, { TextProps } from '../../primitives/Text';

type ChipProps = Omit<TextProps, 'color'> & {
    color?: string;
};

const Chip = ({ children, color, ...rest }: ChipProps) => {
    const theme = useTheme();
    const fontColor = (color || theme.colors.gray) as keyof Theme['colors'];
    const backgroundColor = lightenColor(fontColor, 0.25);
    return (
        <Text
            as="span"
            backgroundColor={backgroundColor}
            borderRadius={12}
            color={fontColor}
            fontSize={12}
            fontWeight="semi-bold"
            lineHeight="12px"
            maxHeight={12}
            paddingBottom={1}
            paddingTop="3px"
            paddingX={2}
            {...rest}
        >
            {children}
        </Text>
    );
};

Chip.displayName = 'Chip';

export { ChipProps };
export default Chip;
