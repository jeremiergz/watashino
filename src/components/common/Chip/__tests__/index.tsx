import React from 'react';
import ReactTestRenderer from 'react-test-renderer';
import { ThemeProvider } from 'styled-components';
import { DarkTheme } from '../../../../theme';
import Chip from '../../Chip';

describe('components/common/Chip Test Suite', () => {
  it('matches snapshot', () => {
    const jsx = (
      <ThemeProvider theme={DarkTheme}>
        <Chip>Chip Component</Chip>
      </ThemeProvider>
    );
    expect(ReactTestRenderer.create(jsx).toJSON()).toMatchSnapshot();
  });
});
