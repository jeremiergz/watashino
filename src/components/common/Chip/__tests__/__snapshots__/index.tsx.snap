// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/widgets/Chip Test Suite matches snapshot 1`] = `
<span
  className="sc-htpNat cTGVaf"
  color="#757575"
  fontSize={12}
  fontWeight="semi-bold"
>
  Chip Component
</span>
`;
